### DISCUSSION TITLE
Proper way to upgrade pixi

### DISCUSSION THREAD

**Discussion by humphd** [prefix-dev/pixi#558] [Q&A]
I've got a large monorepo using Pixi v0.5.0 in a dev container.  Twice I've tried to upgrade to newer versions, and twice I've given up.  I'm hoping to get some advice on how to do this cleanly.

I think the issue might be related to differences in the lock file (a guess) and cached volumes for `.pixi/` dirs in Docker.  The problem I see is that pixi seems to only do a partial install, so later, running tests will suddenly trigger installations, some of which never complete.

Because I have to do this in a dev container that's used by the entire team, I have to do this in a way that will work for everyone when they rebuild for the change on `main`.

Any advice?  I'd like to get on the latest version.

**Comment by ruben-arts** [comment]
There is nothing that comes to mind why that would not work. Could you see if it would be fixed by either deleting the `.pixi` folder and or the `pixi.lock`. Just to find which one is the culprit.

**Reply by humphd** [reply]
Deleting the lock files (I have two, one in root, one deeper in monorepo) would be the ideal, since the dev container has the `.pixi/` dirs in volumes that survive between start/stop/rebuild of the container.  You're saying that blowing away the lock file, but leaving the `.pixi/` dirs in place *should* be OK?

**Reply by ruben-arts** [reply]
That will upgrade the dependencies so if that would be a problem you should specify some more packages on a specific version. But it is hard for me to understand the culprit of the problem without being able to reproduce.

That said, it might be the mounted volumes that don't play nice but to know that for sure we should have a reproducible example.

**Reply by humphd** [reply]
Sure, I appreciate that it's hard to diagnose without STR.  This project isn't public, so I can't point you at it; but I'll experiment with both options and see what does and doesn't work.

Thanks for helping me think it through.

**Comment by lucascolley** [comment]
https://pixi.sh/latest/reference/cli/pixi/upgrade/
